lib/OpenQA/Resource/Locks.pm

```diff
@@ -97,7 +91,7 @@ sub create {
 ## Barriers
 # barriers are created with number of expected jobs. Then wait call waits until the expected number of jobs is waiting
 
-sub barrier_create ($name, $jobid, $expected_jobs) {
+sub barrier_create ($name = undef, $jobid = undef, $expected_jobs = undef) {
```

Comment:
hm, why do we have `return 0 unless $name && $jobid && $expected_jobs;` in here then?

Reply:
Because they might be `undef`. There's a difference between an argument being not passed at all (not the case here) and `undef` being passed as argument (possibly the case here).